File: isilibrary/src/main/java/com/isi/isilibrary/products/Rates.java
package com.isi.isilibrary.products;

public class Rates {

    public static final String A = "A";
    public static final String B = "B";
    public static final String C = "C";
    public static final String N1 = "N1";
    public static final String N2 = "N2";
    public static final String N3 = "N3";
    public static final String N4 = "N4";
    public static final String N5 = "N5";
    public static final String N6 = "N6";
    public static final String N7 = "N7";

    public static final String[] rates = {A,B,C,N1,N2,N3,N4,N5,N6,N7};

    public static int getPRINTFCode(String code){

        switch (code){

            case A:
                return 1;
            case B:
                return 2;
            case N1:
                return 8;
            case N2:
                return 9;
            case N3:
                return 10;
            case N4:
                return 0;
            case N5:
                return 11;
            case N6:
                return 12;
            default:
                return 3;


        }

    }

    public static String getRatesValor(String code){

        switch (code){

            case A:
                return "4%";
            case B:
                return "10%";
            case N1:
                return N1;
            case N2:
                return N2;
            case N3:
                return N3;
            case N4:
                return N4;
            case N5:
                return N5;
            case N6:
                return N6;
            default:
                return "22%";


        }

    }


}

File: isilibrary/src/main/java/com/isi/isilibrary/products/Rates.kt
package com.isi.isilibrary.products

object Rates {
    const val A = "A"
    const val B = "B"
    const val C = "C"
    const val N1 = "N1"
    const val N2 = "N2"
    const val N3 = "N3"
    const val N4 = "N4"
    const val N5 = "N5"
    const val N6 = "N6"
    const val N7 = "N7"
    val rates = arrayOf(A, B, C, N1, N2, N3, N4, N5, N6, N7)
    fun getPRINTFCode(code: String?): Int {
        return when (code) {
            A -> 1
            B -> 2
            N1 -> 8
            N2 -> 9
            N3 -> 10
            N4 -> 0
            N5 -> 11
            N6 -> 12
            else -> 3
        }
    }

    fun getRatesValor(code: String?): String {
        return when (code) {
            A -> "4%"
            B -> "10%"
            N1 -> N1
            N2 -> N2
            N3 -> N3
            N4 -> N4
            N5 -> N5
            N6 -> N6
            else -> "22%"
        }
    }
}